devicetypes/stelpro/stelpro-ki-zigbee-thermostat.src/stelpro-ki-zigbee-thermostat.groovy

```diff
@@ -337,6 +313,45 @@ def parse(String description) {
 	return result
 }
 
+def handleTemperature(descMap) {
+	def map = [:]
+	def intVal = Integer.parseInt(descMap.value, 16)
+	map.name = "temperature"
+	map.unit = getTemperatureScale()
+	map.value = getTemperature(descMap.value)
+
+	// Handle special temperature flags where we need to change the event type
+	if (intVal == 0x7ffd) {
+		map.name = "temperatureAlarm"
+		map.value = "freeze"
+		map.unit = null
+	} else if (intVal == 0x7fff) {
+		map.name = "temperatureAlarm"
+		map.value = "heat"
+		map.unit = null
+	} else if (intVal == 0x8000) {
+		map.name = null
+		map.value = null
+		map.unit = null
+		map.descriptionText = "Received a temperature error"
+	} else if (intVal > 0x8000) {
+		map.value = -(Math.round(2*(655.36 - map.value))/2)
```

Comment:
You should probably leave a comment explaining this special case. Also you probably want to be using `intVal` here instead of the converted `map.value`.